src/vpn-api/app.ts

```diff
@@ -35,34 +39,103 @@ import worker from './worker';
 
 const VPN_INSTANCE_COUNT =
 	parseInt(process.env.VPN_INSTANCE_COUNT!, 10) || os.cpus().length;
+const VPN_BYTECOUNT_INTERVAL =
+	parseInt(process.env.VPN_BYTECOUNT_INTERVAL!, 10) || 0;
 
 if (cluster.isMaster) {
 	logger.info(
 		`open-balena-vpn@${VERSION} master process started with pid ${process.pid}`,
 	);
-	if (VPN_INSTANCE_COUNT > 1) {
-		logger.info(`spawning ${VPN_INSTANCE_COUNT} workers`);
-		_.times(VPN_INSTANCE_COUNT, i => {
-			const instanceId = i + 1;
-			const restartWorker = (code?: number, signal?: string) => {
-				if (signal != null) {
-					logger.error(
-						`open-balena-vpn worker-${instanceId} killed with signal ${signal}`,
-					);
-				}
-				if (code != null) {
-					logger.error(
-						`open-balena-vpn worker-${instanceId} exited with code ${code}`,
-					);
-				}
-				cluster.fork({ VPN_INSTANCE_ID: instanceId }).on('exit', restartWorker);
-			};
-			restartWorker();
+	logger.info(`spawning ${VPN_INSTANCE_COUNT} workers`);
+	_.times(VPN_INSTANCE_COUNT, i => {
+		const instanceId = i + 1;
+		const restartWorker = (code?: number, signal?: string) => {
+			if (signal != null) {
+				logger.error(
+					`open-balena-vpn worker-${instanceId} killed with signal ${signal}`,
+				);
+			}
+			if (code != null) {
+				logger.error(
+					`open-balena-vpn worker-${instanceId} exited with code ${code}`,
+				);
+			}
+			cluster.fork({ VPN_INSTANCE_ID: instanceId }).on('exit', restartWorker);
+		};
+		restartWorker();
+	});
+
+	// track session stats from workers
+	const sessions: {
+		[key: string]: {
+			ts: number;
+			rxBytes: number;
+			txBytes: number;
+		};
+	} = {};
+
+	if (VPN_BYTECOUNT_INTERVAL > 0) {
+		cluster.on('message', (_worker, msg) => {
+			const { uuid, ts, rxBytes, txBytes } = msg;
+			sessions[uuid] = { ts, rxBytes, txBytes };
 		});
 	}
+
+	const kb = 2 ** 10;
```

Comment:
I know this is correct but 1024 would be clearer to me - I don't know for other people though

Reply:
Added a comment.